### DISCUSSION TITLE
Perpendicular Line for Celestial Object

### DISCUSSION THREAD

**Discussion by Vaalarivu** [Stellarium/stellarium#3646] [Ideas]
Hi Stellarium Team, 

I need to construct a perpendicular 90 Degree line with Ecliptic (J2000) line as base. This perpendicular line should also connect the center point of planets / moon so that we can make accurate marking when the perpendicular line moves and touches along the starts that are has high altitude. Kindly check and update on this requirement.

**Comment by gzotti** [comment]
Are you sure it should indicate J2000 longitudes and not ecliptic longitude of date? 
Do you need 1 line with selectable target (limited to Moon/planets), or lines to all planets at the same time, or a single line between (any) currently selected object and ecliptic? 
I don't fully understand the last sentence about star(t?)s. Should the line go from ecliptic pole 90° through the planet to the ecliptic, or even 180° to the other pole, so that you can check planetary conjunctions in ecliptical longitude with fixed sta**rs** in high ecliptical **LA**titude?

**Reply by Vaalarivu** [reply]
Dear zott, Thank you for the quick response.

This is my requirement. 

<img width="810" alt="Screen1" src="https://github.com/Stellarium/stellarium/assets/160964389/9b54177c-eed9-4a9d-939e-0545cbb847af">

- In the above image shown I have selected a star. I want the exact degree of the selected star using the galacticJ2000 line (which shows degree in the above image **REDLINE** )

- **I drew a line YELLOW to mark degree of the star**. the above selected star is near 140Degree. If we have a option to draw a line / scale it would be great!

- Whenever we select an object (eg: Moon) : It should also show degree so that i can move the moon matching to the 140Degree of the star. Which form an alignment. Also if we can have time information added just next to the object it would be much more helpful.

- **Important:** How can i pull report on the Date and Time of the moon entering a constellation? do we option for that ? Kindly suggest and help!

This is the current requirement my lord!
I would like to give more value adds that can be added to the application in the coming days.

warm regards!
Vaalarivu
vaalarivu@gmail.com

**Reply by gzotti** [reply]
OK. It's definitely not galacticJ2000. But I still don't understand why you need a culturally arbitrary ecliptic J2000 instead of ecliptic of date (or even a traditional Indian sidereal ecliptic degree labeling that starts with some star (sorry, I forgot which!), does not start at the equator/ecliptic intersection by definition, and undergoes precession with the stars). Is that a "modernized" Vedic astrology invented or reformulated into the J2000 frame by somebody past 1980?

I can possibly add some features like the line connecting Moon to ecliptic and a line from current object to ecliptic, to the ArchaeoLines plugin, if I understand why, i.e., I would like to add a credible reference with information about culture and application (Vedic astronomy/astrology?), not just because someone in this forum asked for it. The explanation (~1/2 page or more, with credible reference) should go into the User Guide, the explanation author's name shall be attached. 

However, ecliptic coordinates of the target object are given in its info text in upper-left, and time is given in the status bar and time panel, this should be enough.

Stellarium is not the right tool for calculating the "entering times" of the Moon into constellations, signs or other regions. It just provides positional information of the "current" time. You could create a script with minute-to-minute progress (depending on whether full minutes are enough) that stops (or writes output somewhere into a file) when the current constellation/sign/nakshatra changes.
